packages/api-headless-cms-aco/src/hooks/entry/onEntryBeforeRestoreFromBin.hook.ts

```diff
@@ -0,0 +1,37 @@
+import { HcmsAcoContext } from "~/types";
+import WebinyError from "@webiny/error";
+import { ROOT_FOLDER } from "@webiny/api-headless-cms/constants";
+
+export const onEntryBeforeRestoreFromBinHook = (context: HcmsAcoContext) => {
+    const { aco, cms } = context;
+
+    cms.onEntryBeforeRestoreFromBin.subscribe(async ({ entry }) => {
```

Comment:
This hook looks like it has old code (prev logic), am I missing smt here?

Like, we're setting `entry.location.folderId` to `ROOT_FOLDER` in case the folder does not exist.

🤔

Reply:
Exactly! I decoupled the 2 pieces of logic. On entry restoration:

- by default we store back the original `folderId`
- after that, we check if the folder assigned to the entry still exists.

So the logic declared here (`onEntryBeforeRestoreFromBin.hook.ts`) does not change, we changed the way we store the location when we delete an entry.